### ISSUE TITLE
Implement rate limiting middleware to prevent attacks

### ISSUE THREAD

**Issue by colinskow** [colinskow/superlogin#59]
Routes like `/login` are already protected, but request spamming could compromise security in other ways such as with password reset or run up a high bill with the CouchDB provider. I would like to implement rate limiting middleware such as [express-brute](https://github.com/AdamPflug/express-brute) in order to mitigate such risks.

**Comment by tohagan** [comment]
Absolutely!  I also figure that checkUsername and checkEMail will be prime candidates that need exponential rate limiting. non-trivial for these two.

**Comment by tohagan** [comment]
You might also be interested in this one: https://github.com/tj/node-ratelimiter
